src/internal/operations/backup_test.go

```diff
@@ -31,7 +29,272 @@ import (
+	context.Context,
+	model.Schema,
+	manifest.ID,
+	model.Model,
+) error {
+	return errors.New("not implemented")
+}
+
+func (mbs mockBackupStorer) Put(context.Context, model.Schema, model.Model) error {
+	return errors.New("not implemented")
+}
+
+func (mbs mockBackupStorer) Update(context.Context, model.Schema, model.Model) error {
+	return errors.New("not implemented")
+}
+
+// ---------------------------------------------------------------------------
+// helper funcs
+// ---------------------------------------------------------------------------
+
+func makeMetadataPath(
+	t *testing.T,
+	tenant string,
+	service path.ServiceType,
+	resourceOwner string,
+	category path.CategoryType,
+	fileName string,
+) path.Path {
+	t.Helper()
+
+	p, err := path.Builder{}.Append(fileName).ToServiceCategoryMetadataPath(
+		tenant,
+		resourceOwner,
+		service,
+		category,
+		true,
+	)
+	require.NoError(t, err)
+
+	return p
+}
+
+func makeFolderEntry(
+	t *testing.T,
+	pb *path.Builder,
+	size int64,
+	modTime time.Time,
+) *details.DetailsEntry {
+	t.Helper()
+
+	return &details.DetailsEntry{
+		RepoRef:   pb.String(),
+		ShortRef:  pb.ShortRef(),
+		ParentRef: pb.Dir().ShortRef(),
+		ItemInfo: details.ItemInfo{
+			Folder: &details.FolderInfo{
+				ItemType:    details.FolderItem,
+				DisplayName: pb.Elements()[len(pb.Elements())-1],
+				Modified:    modTime,
+				Size:        size,
+			},
+		},
+	}
+}
+
+// TODO(ashmrtn): Really need to factor a function like this out into some
+// common file that is only compiled for tests.
+func makePath(t *testing.T, elements []string, isItem bool) path.Path {
+	t.Helper()
+
+	p, err := path.FromDataLayerPath(stdpath.Join(elements...), isItem)
+	require.NoError(t, err)
+
+	return p
+}
+
+func makeDetailsEntry(
+	t *testing.T,
+	p path.Path,
+	size int,
+	updated bool,
+) *details.DetailsEntry {
+	t.Helper()
+
+	res := &details.DetailsEntry{
+		RepoRef:   p.String(),
+		ShortRef:  p.ShortRef(),
+		ParentRef: p.ToBuilder().Dir().ShortRef(),
+		ItemInfo:  details.ItemInfo{},
+		Updated:   updated,
+	}
+
+	switch p.Service() {
+	case path.ExchangeService:
+		if p.Category() != path.EmailCategory {
+			assert.FailNowf(
+				t,
+				"category %s not supported in helper function",
+				p.Category().String(),
+			)
+		}
+
+		res.Exchange = &details.ExchangeInfo{
+			ItemType: details.ExchangeMail,
+			Size:     int64(size),
+		}
+
+	case path.OneDriveService:
+		parent, err := path.GetDriveFolderPath(p)
+		require.NoError(t, err)
+
+		res.OneDrive = &details.OneDriveInfo{
+			ItemType:   details.OneDriveItem,
+			ParentPath: parent,
+			Size:       int64(size),
+		}
+
+	default:
+		assert.FailNowf(
+			t,
+			"service %s not supported in helper function",
+			p.Service().String(),
+		)
+	}
+
+	return res
+}
+
+func makeManifest(t *testing.T, backupID model.StableID, incompleteReason string) *snapshot.Manifest {
+	t.Helper()
+
+	backupIDTagKey, _ := kopia.MakeTagKV(kopia.TagBackupID)
```

Comment:
What variable is being swallowed?

Reply:
The tag value generated by the func.  Also, this code is not part of the current PR.